core/controllers/domain_objects_validator.py

```diff
@@ -27,10 +27,25 @@
 from core.domain import config_domain
 from core.domain import exp_domain
 from core.domain import state_domain
+from core.domain import story_domain
 
 from typing import Dict, Optional, Union
 
 
+def validate_story_change(obj):
+    """Validates story change.
+
+    Args:
+        obj: dict. Data that needs to be validated.
+
+    Returns:
+        StoryChange. Returns a StoryChange object.
+    """
+    # No explicit call to validate_dict method is necessary, because
+    # StoryChange calls validate method while initialization.
+    return story_domain.StoryChange(obj)
```

Comment:
Hey, instead of doing this, can you introduce `from_dict` into the `BaseChange` that will create the object and then in the schema you can use `'object_class'` instead? 

Also, can you do the same for `validate_exploration_change` and `validate_collection_change`?

Reply:
I tried writing a `from_dict` by referencing the `__init__` method and other implementations of `from_dict`. Could you give this a quick read-over to verify that I'm going in right direction here?

```
@classmethod
    def from_dict(cls, change_dict):
        """Returns a BaseChange domain object representing change dict.
        
        Returns:
            BaseChange. An object containing all changes mapped from
                change_dict.
        """
        cls.validate_dict(change_dict)

        cmd_name = change_dict['cmd']
        cls.cmd = cmd_name

        all_allowed_commands = (
            cls.ALLOWED_COMMANDS + 
            cls.COMMON_ALLOWED_COMMANDS)

        cmd_attribute_names = []
        for cmd in all_allowed_commands:
            if cmd['name'] == cmd_name:
                cmd_attribute_names = (
                    cmd['required_attribute_names'] + cmd[
                        'optional_attribute_names'])
                break

        for attribute_name in cmd_attribute_names:
            setattr(cls, attribute_name, change_dict.get(attribute_name))

        return cls
```

Reply:
After debugging extensively and passing `story_domain.StoryChange` as the `object_class` for `EditableStoryDataHandler`, I ran into schema validation errors as shown below:

![image](https://user-images.githubusercontent.com/44935091/145729486-d94e3f5e-b7b9-439c-88bb-dc7dfba4fc69.png)

I'm successfully passing in `StoryChange` using `from_dict` from `BaseChange`. However, `BaseChange` uses the `validate_dict` method, so using `object_class` in the schema will not work, according to given documentation. I believe that using `validation_method` along with my implementation of `validate_story_change` is the correct approach. I have attached the documentation from `schema_utils.py` that states this and [the wiki cites this as well.](https://github.com/oppia/oppia/wiki/Schemas#case-2-validation_method-key)

![image](https://user-images.githubusercontent.com/44935091/145729598-9c9b72a8-2ddb-46f5-bd1f-9d91f257778b.png)

Reply:
@vojtechjelinek

Reply:
Can we introduce `validate` function for the `BaseChange`, it will call `to_dict` on itself and then run `validate_dict`.